### PR TITLE
feat: consume entity change events to update cache in eds cached client

### PR DESCRIPTION
**Pull request by kishansairam9**

### CODE REVIEW

File: hypertrace-trace-enricher/hypertrace-trace-enricher-impl/src/main/java/org/hypertrace/traceenricher/enrichment/clients/DefaultClientRegistry.java [hunk 1]
```diff
@@ -156,6 +178,14 @@ public AttributeServiceCachedClient getAttributeClient() {
...
     this.grpcChannelRegistry.shutdown();
+    this.entityChangeEventListener.ifPresent(
+        listener -> {
+          try {
+            listener.close();
+          } catch (Exception e) {
+
+          }
+        });
   }
```
Comment (kishansairam9) [inline hypertrace-trace-enricher/hypertrace-trace-enricher-impl/src/main/java/org/hypertrace/traceenricher/enrichment/clients/DefaultClientRegistry.java:189]: This method isn't called really, but added the close for sanity

File: gradle/libs.versions.toml [hunk 2]
```diff
@@ -37,6 +40,7 @@ slf4j-api = { module = "org.slf4j:slf4j-api", version = "2.0.5" }
 apache-log4j-slf4jImpl = { module = "org.apache.logging.log4j:log4j-slf4j2-impl", version = "2.20.0" }
 apache-commons-lang3 = { module = "org.apache.commons:commons-lang3", version = "3.12.0" }
 apache-kafka-kafkaStreamsTestUtils = { module = "org.apache.kafka:kafka-streams-test-utils", version = "7.2.1-ccs" }
+kafka-protobufserde = { module = "io.confluent:kafka-streams-protobuf-serde", version.ref = "confluent-platform" }
```
Comment (aaron-steinfeld) [inline gradle/libs.versions.toml:43]: kafka versions should be coming from the kafka-bom, should be able to remove the declaration here and in the one above.

File: hypertrace-trace-enricher/hypertrace-trace-enricher-impl/src/main/java/org/hypertrace/traceenricher/enrichment/clients/DefaultClientRegistry.java [hunk 3]
```diff
@@ -165,4 +195,44 @@ protected Channel buildChannel(String host, int port) {
...
+              .build(
+                  consumerName,
+                  clientsConfig.getConfig(ENTITY_CHANGE_EVENTS_CONFIG_KEY),
+                  getEntityChangeEventKeySerde(serdeConfig),
+                  getEntityChangeEventValueSerde(serdeConfig)));
+    }
+    return Optional.empty();
+  }
+
+  private static Deserializer<EntityChangeEventKey> getEntityChangeEventKeySerde(
```
Comment (aaron-steinfeld) [inline hypertrace-trace-enricher/hypertrace-trace-enricher-impl/src/main/java/org/hypertrace/traceenricher/enrichment/clients/DefaultClientRegistry.java:221]: naming - 'serde' refers to serializer/deserializer, when we lump both together. Given you're only working with deserialization here the naming should be adjusted.

**Review by aaron-steinfeld — COMMENTED** [review]
other than naming lgtm

File: hypertrace-trace-enricher/hypertrace-trace-enricher-impl/src/main/java/org/hypertrace/traceenricher/enrichment/clients/DefaultClientRegistry.java (hunk 3 above)
Comment (aaron-steinfeld) [reply hypertrace-trace-enricher/hypertrace-trace-enricher-impl/src/main/java/org/hypertrace/traceenricher/enrichment/clients/DefaultClientRegistry.java:221]: nit - method name too

**Review by aaron-steinfeld — APPROVED** [review]

**Review by aaron-steinfeld — APPROVED** [review]

**Pull request merged**